File: example/src/main/kotlin/com/xebia/functional/auto/Recipe.kt
package com.xebia.functional.auto

import kotlinx.serialization.Serializable

@Serializable
data class Recipe(val name: String, val ingredients: List<String>)

suspend fun main() {
    val recipe: Recipe = ai("Recipe for chocolate chip cookies.")
    println("The recipe for ${recipe.name} is ${recipe.ingredients}")
}

File: examples/scala/src/main/scala/com/xebia/functional/xef/scala/conversation/serialization/Recipe.scala
package com.xebia.functional.xef.scala.conversation.serialization

import com.xebia.functional.xef.prompt.Prompt
import com.xebia.functional.xef.scala.conversation.*
import io.circe.Decoder

private final case class Recipe(name: String, ingredients: List[String]) derives SerialDescriptor, Decoder

@main def runRecipe: Unit =
  conversation {
    val recipe: Recipe = prompt(Prompt("Recipe for chocolate chip cookies."))
    println(s"The recipe for ${recipe.name} is ${recipe.ingredients}")
  }
